src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,555 @@
+
+        Self{logger,display_object,scene,breadcrumbs,project_name,breadcrumbs_container,
+            frp_inputs,current_index,frp_debug,camera,background}.init()
+    }
+
+    fn init(self) -> Self {
+        self.add_child(&self.project_name);
+        self.add_child(&self.breadcrumbs_container);
+        self.project_name.set_position(Vector3(HORIZONTAL_MARGIN,0.0,0.0));
+        self.relayout_for_project_name_width(self.project_name.width());
+        self.project_name.frp.select.emit(());
+        self
+    }
+
+    fn camera_changed(&self) {
+        let camera     = &self.camera;
+        let screen     = camera.screen();
+        let x_position = -screen.width/2.0;
+        let y_position = screen.height/2.0;
+        self.set_position(Vector3(x_position.round(),y_position.round(),0.0));
+    }
+
+    fn width(&self) -> f32 {
+        self.breadcrumbs.borrow().iter().map(|breadcrumb| breadcrumb.width()).sum()
+    }
+
+    fn relayout_for_project_name_width(&self, width:f32) {
+        self.breadcrumbs_container.set_position_x((HORIZONTAL_MARGIN+width).round());
+    }
+
+    fn get_breadcrumb(&self, index:usize) -> Option<Breadcrumb> {
+        (index > 0).as_option().and_then(|_|
+            self.breadcrumbs.borrow_mut().get(index - 1).map(|breadcrumb| breadcrumb.clone_ref())
+        )
+    }
+
+    /// Selects the breadcrumb identified by its `index` and returns `(popped_count,local_calls)`,
+    /// where `popped_count` is the number of breadcrumbs in the right side of `index` that needs to
+    /// be popped or a list of `LocalCall`s identifying the breadcrumbs we need to push.
+    fn select_breadcrumb(&self, index:usize) -> (usize,Vec<Option<LocalCall>>) {
+        info!(self.logger,"Selecting breadcrumb #{index}.");
+        let current_index = self.current_index.get();
+        match index.cmp(&current_index) {
+            Ordering::Less    => (current_index - index, default()),
+            Ordering::Equal   => default(),
+            Ordering::Greater => {
+                let mut local_calls = Vec::new();
+                for index in current_index..index {
+                    let info = self.breadcrumbs.borrow().get(index).map(|breadcrumb| {
+                        let definition = breadcrumb.info.method_pointer.clone();
+                        let call       = breadcrumb.info.expression_id;
+                        LocalCall{call,definition}
+                    }).as_ref().cloned();
+                    if info.is_some() {
+                        local_calls.push(info);
+                    } else {
+                        error!(self.logger, "LocalCall info is not present.");
+                        self.remove_breadcrumbs_history_beginning_from(index);
+                        break;
+                    }
+                }
+                (default(),local_calls)
+            },
+        }
+    }
+
+    fn push_breadcrumb(&self, local_call:&Option<LocalCall>) -> Option<(usize,usize)> {
```

Comment:
For each type which has a lambda in output type, add at least one sentence of explanation what the output type is please.